python/cudf/cudf/core/column/numerical.py

```diff
@@ -412,6 +413,14 @@ def find_and_replace(
         """
         Return col with *to_replace* replaced with *value*.
         """
+        to_replace_col = as_column(to_replace)
+        replacement_col = as_column(replacement)
+
+        if not isinstance(to_replace_col, NumericalColumn) or not isinstance(
+            replacement_col, NumericalColumn
+        ):
+            return self
```

Comment:
If replacing numbers with e.g.,  strings, should we fail instead?

Reply:
I think we should allow this operation. Reason is that a dataframe that could have a mix of string and numeric values will be called with some generic `to_replace` & `value` params like:

```python
>>> import pandas as pd
>>> df = pd.DataFrame({'a':[1, 2, 3, 4], 'strings':['a','b','d','e']})
>>> df
   a strings
0  1       a
1  2       b
2  3       d
3  4       e
>>> df.replace(1, 111)
     a strings
0  111       a
1    2       b
2    3       d
3    4       e
>>> df.replace("a", "aaa")
   a strings
0  1     aaa
1  2       b
2  3       d
3  4       e
```

In such cases failing with an error message asking the user to drop the rest of the columns and do replace and again the caller of the API will have to perform the reordering/rejoining of the dataframe columns would be painful. Instead, we can simply ignore if we detect mismatching dtypes are passed.

Reply:
Got it. What happens when we do the same with a `Series`? For example:

```python
In [2]: a = cudf.Series([1, 2, 3])

In [3]: a.replace(1, "a")
```

Do we fail or silently return?

Reply:
Return silently in case of series aswell.

Reply:
Pandas actually does the replace here, which makes me think we should either raise or clearly document the difference in behaviour with an example.

Reply:
Added an example and mentioned the difference in behavior with respect to pandas.

Reply:
FWIW I think we should throw in this situation. If the dtype of `to_replace` and `value` aren't the same with reason (allow different integers / floats for example), we're in a situation we don't want to support in cuDF and we should throw accordingly.

Reply:
Got it, so we want to error loudly for:

```python
>>> s = cudf.Series(['a', 'b', 'c'])
>>> s.replace(['a', 'b'], [0, 1])
```

and return silently for:
```python
>>> s = cudf.Series(['a', 'b', 'c'])
>>> s.replace([1, 2], [3, 4])
```

Did I get it right?

Reply:
Yup, those are my thoughts at least. What do you think @galipremsagar and @shwina?

Reply:
I think it makes sense to me. I have added the validation and tests validating them.